.github/workflows/release.yml

```diff
@@ -1,42 +1,31 @@
 name: Release
-
+# This workflow helps with creating releases.
+# This job will only be triggered when a tag (vX.X.x) is pushed
 on:
   push:
+    # Sequence of patterns matched against refs/tags
     tags:
-      - '*'
+      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
+
+permissions:
+  contents: read
 
 jobs:
-  buildx:
+  release:
+    permissions:
+      contents: write # for goreleaser/goreleaser-action to create a GitHub release
     runs-on: ubuntu-latest
     steps:
-      -
-        name: Checkout
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+      - name: Install Go
+        uses: actions/setup-go@v3
         with:
-          fetch-depth: 0
-      -
-        name: Fetch all tags
-        run: git fetch --force --tags
-
-      - name: Set up QEMU
-        id: qemu
-        uses: docker/setup-qemu-action@v2
-
-      -
-        name: Use Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v2
-      -
-        name: Build image for release
```

Comment:
Whats the impact of removing the images for release here?

Reply:
This job uses go-releaser directly to build and publish releases. Not sure why Docker was used...